tests/main/refresh-delta-from-core/task.yaml

```diff
@@ -0,0 +1,23 @@
+summary: Check refresh works with xdelta3 from the core snap
+
+# delta downloads are currently disabled by default on core
+systems: [-ubuntu-core-*]
+
+environment:
+    SNAP_NAME: test-snapd-delta-refresh
+
+prepare: |
+    echo "Ensure no xdelta3 is installed on the host"
+    apt-get remove -y xdelta3
+
+    echo "Given a snap is installed"
+    snap install --edge $SNAP_NAME
+
+restore: |
+    apt-get install -y xdelta3
```

Comment:
Restore and prepare are not strictly speaking symmetrical. I think we should have a helper function like:

```
. $TESTLIB/packages.sh
syspkg_mark
syspkg_install xdelta3

...

syspkg_revert_to_mark
```

We could do mark/revert in the generic code so that tests don't have a burden of correctly removing packages each time.